src/hssm/hssm.py

```diff
@@ -975,27 +975,39 @@ def plot_trace(
             Whether to call plt.tight_layout() after plotting. Defaults to True.
         """
         data = data or self.traces
+        assert isinstance(data, az.InferenceData)
 
-        if not include_deterministic:
-            var_names = self._get_deterministic_var_names(data)
-            if var_names:
-                if "var_names" in kwargs:
-                    if isinstance(kwargs["var_names"], str):
-                        if kwargs["var_names"] not in var_names:
-                            var_names.append(kwargs["var_names"])
-                        kwargs["var_names"] = var_names
-                    elif isinstance(kwargs["var_names"], list):
-                        kwargs["var_names"] = list(
-                            set(var_names) | set(kwargs["var_names"])
-                        )
-                    elif kwargs["var_names"] is None:
-                        kwargs["var_names"] = var_names
+        if data is not None:
+            if not include_deterministic:
+                var_names = list(
+                    set([var.name for var in self.pymc_model.free_RVs]).intersection(
+                        set(list(data["posterior"].data_vars.keys()))
+                    )
+                )
+                # var_names = self._get_deterministic_var_names(data)
+                if var_names:
+                    if "var_names" in kwargs:
+                        if isinstance(kwargs["var_names"], str):
+                            if kwargs["var_names"] not in var_names:
+                                var_names.append(kwargs["var_names"])
+                            kwargs["var_names"] = var_names
+                        elif isinstance(kwargs["var_names"], list):
+                            kwargs["var_names"] = list(
+                                set(var_names) | set(kwargs["var_names"])
+                            )
+                        elif kwargs["var_names"] is None:
+                            kwargs["var_names"] = var_names
+                        else:
+                            raise ValueError(
+                                "`var_names` must be a string, a list of strings"
+                                ", or None."
+                            )
                     else:
-                        raise ValueError(
-                            "`var_names` must be a string, a list of strings, or None."
-                        )
-                else:
-                    kwargs["var_names"] = var_names
+                        kwargs["var_names"] = var_names
+        elif (not isinstance(data, az.InferenceData)) and (data is not None):
+            raise ValueError("data must be an InferenceData object.")
+        elif data is None:
+            raise ValueError("Please sample to model first.")
```

Comment:
For clarity, I'd use just one if block. We already have an assert statement to ensure that `data` is of type `az.InferenceData`, so just an else block that throws an error should be sufficient

Reply:
Ok thinking about the logic, I think we were able to just take the whole if logic out and rely on the fact that `data` must be `InferenceData` at this point anyways. @digicosmos86

Reply:
This needed to change in two places.
Will let the tests run now, then merge?